### ISSUE TITLE
push docker image fails with AWS CLI v2

### ISSUE THREAD

**Issue by gillouche** [Kenza-AI/sagify#117]
Hello,

in the push.sh template file, we try to login to ECR with 
```bash
aws ecr get-login --profile ${profile} --region ${region} --no-include-email
```
This does not work with AWS CLI v2. The correct command is
```bash
aws --region <region> ecr get-login-password --profile <profile> | docker login --username AWS --password-stdin <aws account number>.dkr.ecr.<region>.amazonaws.com
```

Please see this [issue](https://github.com/aws/aws-cli/issues/4962) for more information.

**Comment by ilazakis** [comment]
Hey @gillouche , thanks for raising the issue; I haven’t upgraded to cli v2 yet, thank you for pointing out the corresponding issue on AWS’s repo.

Sounds like we’re going to have to adjust the command to accommodate for both versions if possible (presumably by checking the aws cli’s version before talking to ECR).

**Comment by juliensimon** [comment]
Hi,

I hit the same issue. Here's the fix for push.sh.

```
# Get the login command from ECR and execute it directly
if [[ ! -z "$profile" ]]; then
     $(aws --region ${region} ecr get-login-password --profile ${profile} | docker login --username AWS --password-stdin ${account}.dkr.ecr.${region}.amazonaws.com)
else
     $(aws --region ${region} ecr get-login-password | docker login --username AWS --password-stdin ${account}.dkr.ecr.${region}.amazonaws.com)
fi
```
AWS CLI v2 has been out for almost 6 months now, I'd recommend going for it :)

**Comment by makarono** [comment]
Hi

this is my oneliner which works . You just need to change PROF variable value to your aws cli profile id

PROF=test-poc; $(aws ecr get-login-password --profile ${PROF} | docker login -u AWS --password-stdin $(aws sts get-caller-identity --query 'Account' --output text --profile ${PROF}).dkr.ecr.$(aws configure get region --profile ${PROF}).amazonaws.com)

**Comment by pm3310** [comment]
Hey folks, @makarono @gillouche @juliensimon issue should be solved since Sagify version 0.20.8. FYI, the lastest Sagify version is 0.20.9.
